pkg/chunk/gcp/bigtable_index_client.go

```diff
@@ -179,7 +197,10 @@ func (s *storageClientColumnKey) BatchWrite(ctx context.Context, batch chunk.Wri
 			rowKeys = append(rowKeys, rowKey)
 			muts = append(muts, mut)
 		}
-
+		if s.writeLimiter.Allow() == false {
```

Comment:
`Allow()` is a shorthand for `AllowN(time.Now(), 1)`, which increments by 1 the number of "consumed" tokens in the rate limiter. Using it here is not correct, because we don't want to increment by 1 the number of tokens, and then increment again in the `WaitN()`.

I would suggest to remove the `Allow()` at all, and just call the `WaitN()`. You will not be able to track your `rateLimitedWriteRequest` metric, but I don't see any non-hacky wait to do it.

Reply:
I got it, would have been great if we could track metric with waitN.